apps/src/lab2/views/components/Instructions.tsx

```diff
@@ -47,15 +54,26 @@ const Instructions: React.FunctionComponent<InstructionsProps> = props => {
   );
   const levelIndex = useSelector(currentLevelIndex);
   const currentLevelCount = useSelector(levelCount);
-  const validationState = useSelector(
+  const {hasConditions, message, satisfied} = useSelector(
     (state: {lab: LabState}) => state.lab.validationState
   );
+
+  // If there are no validation conditions, we can show the next button so long as
+  // there is another level. If validation is present, also check that conditions are satisfied.
   const showNextButton =
```

Comment:
Just curious - for the last level in a progression, will the bubble turn green for the user at some point? Or is validation needed?

Reply:
Good question! This is actually a case we don't handle in Lab2 yet, so I'm not sure. Looks like there is a lot of code in legacy labs that handles the finish/continue buttons, so we'll need to see what makes sense to adapt into Lab2. Filed https://codedotorg.atlassian.net/browse/LABS-36 to investigate.